### PR TITLE
Fix things getting marked as dirty during initialization

### PR DESCRIPTION
**Pull request by markedmondson**
### What is the purpose of this pull request?
Overwrite more dirty methods. I tried several ways to get to the underlying database_mutations or resetting the attributes after defaults had been set but this was just the easiest sadly.

### What changes did you make? (overview)
Add more overwritten dirty methods

### Is there anything you'd like reviewers to focus on?
Tests

### Checklist

- [x] I've added tests for this change
- [ ] I've added a Changelog entry
- [ ] I've updated a documentation (Readme)

### CODE REVIEW

**Comment by markedmondson** [comment]
Hmm nope!

**Comment by palkan** [comment]
Thanks for catching this.

I'll take a closer look in the next couple of days. Will try to dig deeper into the internals and try to find some magic API to solve all the problems)

**Comment by markedmondson** [comment]
Let me know how you get on, otherwise happy to fix the Rails 5 error. It is getting a bit messy with the full re-write of Dirty methods though :(

**Comment by palkan** [comment]
> otherwise happy to fix the Rails 5 error

I'm thinking about dropping dirty tracking support for Rails 5 completely 🙂 So, don't worry about it)

File: spec/cases/store_attribute_spec.rb [hunk 1]
```diff
@@ -319,11 +319,25 @@
...
+    it "should not mark attributes as dirty after reading original values" do
       reloaded_user = User.take
       reloaded_user.inspect
 
       expect(reloaded_user.changes).to eq({})
+      expect(reloaded_user.changed_attributes).to eq({})
+      expect(reloaded_user.changed?).to be false
+    end
+
+    it "should not mark attributes as dirty after default values are set" do
```
Comment (palkan) [inline spec/cases/store_attribute_spec.rb:331]: Rails adds defaults to changes when they're first set (I tried with non-store attributes here https://github.com/palkan/store_attribute/pull/24)

**Review by palkan — COMMENTED** [review]
I ended up with a huge refactoring here: https://github.com/palkan/store_attribute/pull/24

No more Rails internal duplication: accessors, dirty tracking methods, prefixes are provided by Rails. We just do our job — typecasting data.
 
All (well, but one) existing tests pass.

(Left a comment regarding new record changes and defaults).

**Pull request closed without merging**